### PR TITLE
DBT-736 Lessons Calendar Add Lesson

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

We are implementing a new feature of adding and viewing lessons as well as adding students and materials to it

[Lessons Material Lists and Lessons Add Material](https://www.notion.so/tianlu/Frontend-Lessons-Material-Lists-and-Lessons-Add-Material-9f052e953bf741c8a9aa38c2b4fc4339?pvs=4)

## Test

- Login as Admin
- in the sidebar look for the handle lesson and click on it

## Platform
Google Chrome

## Proof of Concept
image

### CODE REVIEW

File: src/modules/lessons/calendar-lessons-list/components/calendar-lessons-list.vue [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        <v-col cols="auto">
+          <div><span class="font-weight-bold subtitle-2">Fecha:</span>03 August 2023</div>
+        </v-col>
+        <v-col cols="auto">
+          <div><span class="font-weight-bold subtitle-2">tiempo:</span>09:00 - 10:00</div>
+        </v-col>
+        <v-col cols="auto">
+          <div><span class="font-weight-bold subtitle-2">Estudiantes totales:</span>245</div>
+        </v-col>
+      </v-row>
```
Comment (abelbordonado) [inline src/modules/lessons/calendar-lessons-list/components/calendar-lessons-list.vue:28]: This feels to be another component

File: src/modules/lessons/calendar-lessons-list/components/calendar-lessons-list.vue [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+      <v-spacer />
```
Comment (abelbordonado) [inline src/modules/lessons/calendar-lessons-list/components/calendar-lessons-list.vue:30]: This spacer wont go on this component but on the parent

File: src/modules/lessons/calendar-lessons-list/components/calendar-lessons-list.vue [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        <v-col cols="auto">
+          <div class=""><span class="font-weight-bold subtitle-2">Lession:</span> El nombre de la lección va aquí</div>
+        </v-col>
+        <v-col cols="auto">
+          <div><span class="font-weight-bold subtitle-2">Fecha:</span>03 August 2023</div>
+        </v-col>
+        <v-col cols="auto">
+          <div><span class="font-weight-bold subtitle-2">tiempo:</span>09:00 - 10:00</div>
+        </v-col>
+        <v-col cols="auto">
```
Comment (abelbordonado) [inline src/modules/lessons/calendar-lessons-list/components/calendar-lessons-list.vue:25]: This toolbar shall be another component

File: src/modules/lessons/calendar-lessons-list/lessons.store.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import DataTableStore from '@/modules/resources/store/data-table.store'
```
Comment (abelbordonado) [inline src/modules/lessons/calendar-lessons-list/lessons.store.js:1]: Do we? 
This page has no pagination neither search :D may not be needed

File: src/modules/lessons/create-lesson/components/CommentFieldInput.vue [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      type: String,
+      default: ''
+    },
+    rules: {
+      type: [Object, String],
+      required: true
+    }
+  }
+}
+</script>
```
Comment (abelbordonado) [inline src/modules/lessons/create-lesson/components/CommentFieldInput.vue:32]: Seems overkilling for a component.

I think we have a generic FieldINut which can be use straight away without this layer of abstraction.  Same for NameInput

File: src/modules/resources/components/form/FieldInput.vue [hunk 6]
```diff
@@ -59,7 +59,13 @@ export default {
       this.update(value)
     }
   },
+  mounted() {
+    this.resetErrors()
+  },
   methods: {
+    resetErrors() {
+      this.$refs.validationProvider.reset()
+    },
```
Comment (abelbordonado) [inline src/modules/resources/components/form/FieldInput.vue:68]: Thats good

File: src/modules/resources/components/form/input.vue [hunk 7]
```diff
@@ -10,28 +11,37 @@
...
-export default {
-  name: 'NombreFieldInput',
-  props: {
-    value: {
-      type: String,
-      default: ''
+  
+  <script>
+  export default {
+    name: 'NombreFieldInput',
```
Comment (abelbordonado) [inline src/modules/resources/components/form/input.vue:26]: Old issue, lets change this to `input` (was copy pasted in another branch)

File: src/modules/resources/components/form/student-auto-complete.vue [hunk 8]
```diff
@@ -33,7 +33,7 @@ export default {
   props: {
     limit: {
       type: Number,
-      default: 5
+      default: 3
```
Comment (abelbordonado) [inline src/modules/resources/components/form/student-auto-complete.vue:36]: ?? Lol 

wrong

Pass just a prop for the specific use has more sense. 

<AutoComplete limit="3" />

File: src/modules/resources/components/form/tags-auto-complete.vue [hunk 9]
```diff
@@ -0,0 +1,86 @@
+<template>
```
Comment (abelbordonado) [inline src/modules/resources/components/form/tags-auto-complete.vue:1]: May have conflict here, I had added the validator in another branch, will see given the time

File: src/modules/lessons/create-lesson/components/CommentFieldInput.vue (hunk 5 above)
Comment (kayiganwa04) [reply src/modules/lessons/create-lesson/components/CommentFieldInput.vue:32]: components here are different this one is a text area and the other one is input

File: src/modules/resources/components/form/student-auto-complete.vue (hunk 8 above)
Comment (kayiganwa04) [reply src/modules/resources/components/form/student-auto-complete.vue:36]: am not sure how this was touched but it is not supposed to be touched in this task

**Pull request merged**